packages/typespec-client-generator-core/src/internal-utils.ts

```diff
@@ -257,9 +260,69 @@ export function getSdkTypeBaseHelper<TKind>(
     __raw: type,
     deprecation: getDeprecationDetails(context.program, type)?.message,
     kind,
+    decorators: getTypeDecorators(context, type),
   };
 }
 
+export function getTypeDecorators(context: TCGCContext, type: Type): Record<string, Array<any>> {
+  const retval: Record<string, Array<any>> = {};
+  if ("decorators" in type) {
+    for (const decorator of type.decorators) {
+      // only process explicitly defined decorators
+      if (decorator.definition) {
+        const decoratorName = `${decorator.definition?.namespace ? getNamespaceFullName(decorator.definition.namespace) + "." : ""}${decorator.definition?.name}`;
+        // white list filtering
+        if (!context.decoratorsWhiteList) {
+          continue;
+        }
+        let found = false;
+        for (const item of context.decoratorsWhiteList) {
+          if (decoratorName === item) {
+            found = true;
+            break;
+          }
+          const regex = new RegExp(item);
+          if (regex.test(decoratorName)) {
+            found = true;
+            break;
+          }
+        }
+        if (!found) {
+          continue;
+        }
+        retval[decoratorName] = [];
+        for (const arg of decorator.args) {
+          retval[decoratorName].push(getDecoratorArgValue(arg.jsValue));
+        }
+      }
+    }
+  }
+  return retval;
+}
+
+function getDecoratorArgValue(
+  arg:
+    | Type
+    | Record<string, unknown>
+    | Value
+    | unknown[]
+    | string
+    | number
+    | boolean
+    | Numeric
+    | null
+): any {
+  if (typeof arg === "object" && arg !== null && "kind" in arg) {
```

Comment:
I think it would be simpler to type `arg` as a type that includes `{kind: string} | undefined

Reply:
i don't know whether this kind of judgment or use `as` to case to a type to compare kink is better?